src/server.cpp

```diff
@@ -1708,12 +1708,21 @@ void Server::SendSpawnParticle(u16 peer_id, u16 protocol_version,
 {
 	DSTACK(FUNCTION_NAME);
 	if (peer_id == PEER_ID_INEXISTENT) {
-		// This sucks and should be replaced by a better solution in a refactor:
 		std::vector<u16> clients = m_clients.getClientIDs();
+
 		for (std::vector<u16>::iterator i = clients.begin(); i != clients.end(); ++i) {
 			RemotePlayer *player = m_env->getPlayer(*i);
 			if (!player)
 				continue;
+
+			PlayerSAO *sao = player->getPlayerSAO();
+			if (!sao)
+				continue;
+
+			if (sao->getBasePosition().getDistanceFrom(pos) >
+					160.0f)
```

Comment:
@sfan5 suggested to use `active_block_range` instead https://github.com/minetest/minetest/issues/5667#issuecomment-297980867